### PR TITLE
CMDungeon

### PR DESCRIPTION
**Pull request by Ganimoth**
# New Port for CMDungeon

## Game Information
- **Title**: CMDungeon
- **URL**: https://cuca.itch.io/cmdungeon

### CFW Tests
Ensure your game has been tested on all major CFWs:
- [x ] AmberELEC
- [x ] ArkOS
- [x ] ROCKNIX
- [x ] muOS

### Resolution Tests
Test all major resolutions:
- [x ] 480x320
- [x ] 640x480
- [ ] 720x720 (RGB30)
- [x ] Higher resolutions (e.g., 1280x720)

### CODE REVIEW

File: ports/cmdungeon/CMDungeon.sh [hunk 1]
```diff
@@ -0,0 +1,73 @@
...
+source $controlfolder/control.txt
+
+
+[ -f "${controlfolder}/mod_${CFW_NAME}.txt" ] && source "${controlfolder}/mod_${CFW_NAME}.txt"
+
+get_controls
+
+GAMEDIR=/$directory/ports/cmdungeon/
+CONFDIR="$GAMEDIR/conf/"
+
```
Comment (Cebion) [inline ports/cmdungeon/CMDungeon.sh:24]:
```suggestion

> "$GAMEDIR/log.txt" && exec > >(tee "$GAMEDIR/log.txt") 2>&1
```

File: ports/cmdungeon/CMDungeon.sh [hunk 2]
```diff
@@ -0,0 +1,73 @@
...
+# Ensure the conf directory exists
+mkdir -p "$GAMEDIR/conf"
+
+# Set the XDG environment variables for config & savefiles
+export XDG_CONFIG_HOME="$CONFDIR"
```
Comment (Cebion) [inline ports/cmdungeon/CMDungeon.sh:29]:
```suggestion
```

Comment (Cebion) [reply ports/cmdungeon/CMDungeon.sh:29]: Only Data_home is required for XDG Data redirect

File: ports/cmdungeon/CMDungeon.sh (hunk 1 above)
Comment (Cebion) [reply ports/cmdungeon/CMDungeon.sh:24]: Logging missing

File: ports/cmdungeon/CMDungeon.sh [hunk 3]
```diff
@@ -0,0 +1,73 @@
...
+export XDG_DATA_HOME="$CONFDIR"
```
Comment (Cebion) [inline ports/cmdungeon/CMDungeon.sh:30]:
```suggestion
export XDG_DATA_HOME="$CONFDIR"
export SDL_GAMECONTROLLERCONFIG="$sdl_controllerconfig"
```

Comment (Cebion) [reply ports/cmdungeon/CMDungeon.sh:30]: export of SDL_GAMECONTROLLERCONFIG is done here not in the run line

File: ports/cmdungeon/CMDungeon.sh [hunk 4]
```diff
@@ -0,0 +1,73 @@
...
+godot_file="$controlfolder/libs/${runtime}.squashfs"
+$ESUDO mkdir -p "$godot_dir"
+$ESUDO umount "$godot_file" || true
+$ESUDO mount "$godot_file" "$godot_dir"
+PATH="$godot_dir:$PATH"
+
+export FRT_NO_EXIT_SHORTCUTS=FRT_NO_EXIT_SHORTCUTS
+
+$GPTOKEYB "$runtime" -c "cmdungeon.gptk" &
+SDL_GAMECONTROLLERCONFIG="$sdl_controllerconfig" "$runtime" $GODOT_OPTS --main-pack "gamedata/CMDungeon v0.2.0.pck"
```
Comment (Cebion) [inline ports/cmdungeon/CMDungeon.sh:68]:
```suggestion
pm_platform_helper $runtime
"$runtime" $GODOT_OPTS --main-pack "gamedata/CMDungeon v0.2.0.pck"
```

Comment (Cebion) [reply ports/cmdungeon/CMDungeon.sh:68]: pm_platform_helper missing

**Review by Cebion — CHANGES_REQUESTED** [review]

File: ports/cmdungeon/CMDungeon.sh (hunk 4 above)
Comment (Ganimoth) [reply ports/cmdungeon/CMDungeon.sh:68]: Seems I should have waited a bit, I see now you have updated the sample script on the site

**Review by Cebion — APPROVED** [review]

**Review by JanTrueno — APPROVED** [review]
👍

**Pull request merged**